helix-term/src/ui/popup.rs

```diff
@@ -150,7 +156,17 @@ impl<T: Component> Popup<T> {
             rel_x = rel_x.saturating_sub((rel_x + width).saturating_sub(viewport.width));
         }
 
-        let can_put_below = viewport.height > rel_y + MIN_HEIGHT;
+        let can_put_below = if is_menu && editor.menu_border() {
+            // in case of menu borders, we dont want to put it below if it would show two lines less
+            (child_size.1 <= MIN_HEIGHT && viewport.height > rel_y + MIN_HEIGHT)
```

Comment:
J think this is a bit hard to understand. If I understand the code correctly a simpler way to I.plement this would be `let min_height = if is_menu && has_border && child_size.1 > MIN_HEIGHT { MIN_HRIGHT + 2 } else { MIN_HEIGHT };